components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -5025,4 +5044,20 @@ else if(userAgent.contains("MSIE")){
         }
         return userBrowser;
     }
+
+    /**
+     * Url validator, Allow any url with https and http.
+     * Allow any url without fully qualified domain
+     *
+     * @param url Url as string
+     * @return boolean type stating validated or not
+     */
+    private static boolean isURL(String url) {
+
+        Pattern pattern = Pattern.compile("^(http|https)://(.)+", Pattern.CASE_INSENSITIVE);
```

Comment:
Compiling pattern objects within the critical path of a request can be expensive. Consider caching the compiled pattern and re-use. Patterns are thread-safe so, the aforesaid action won't lead to any concurrency issues as well.